engine/config/2.0/schema.json

```diff
@@ -449,63 +449,72 @@
+                  "type": "object",
+                  "properties": {
+                    "host": {
                       "type": "string",
-                      "enum": [
-                        "disable",
-                        "allow",
-                        "prefer",
-                        "require",
-                        "verify-ca",
-                        "verify-full"
+                      "description": "The hostname of the external PostgreSQL server. When set, connects to an external database instead of using embedded mode."
+                    },
+                    "port": {
+                      "description": "The port for the PostgreSQL server. Defaults to 5432.",
+                      "oneOf": [
+                        { "type": "integer", "default": 5432 },
+                        { "$ref": "#/$defs/envVarPattern" },
+                        { "$ref": "#/$defs/filePattern" }
                       ]
                     },
-                    { "$ref": "#/$defs/envVarPattern" },
-                    { "$ref": "#/$defs/filePattern" }
-                  ]
+                    "user": {
+                      "type": "string",
+                      "description": "The username for the PostgreSQL connection. Required for external mode."
+                    },
+                    "password": {
+                      "type": "string",
+                      "description": "The password for the PostgreSQL connection. Required for external mode."
+                    },
+                    "db": {
+                      "type": "string",
+                      "description": "The database name. Required for external mode."
+                    },
+                    "sslmode": {
+                      "description": "The SSL mode for the PostgreSQL connection. Required for external mode.",
+                      "oneOf": [
+                        {
+                          "type": "string",
+                          "enum": [
+                            "disable",
+                            "allow",
+                            "prefer",
+                            "require",
+                            "verify-ca",
+                            "verify-full"
+                          ]
+                        },
+                        { "$ref": "#/$defs/envVarPattern" },
+                        { "$ref": "#/$defs/filePattern" }
+                      ]
+                    }
+                  },
+                  "required": ["host", "port", "user", "password", "db", "sslmode"],
+                  "additionalProperties": false
                 }
-              },
-              "additionalProperties": false
+              ]
             }
           },
           "required": ["postgres"],
           "additionalProperties": false
-        },
-        {
-          "type": "null",
-          "description": "No storage configuration is set. Uses embedded PostgreSQL with default settings."
```

Comment:
Removed this because it was kinda hard to understand. Technically we allowed `storage: null` which meant embedded, but I think it's much clearer to just declare what you have now:
```
storage:
  postgres:
    data_path: "..."
```